syntax = "proto3";

message Introduction {
  string pseudonym = 1;
  string targetInterface = 2;
  bytes nonce = 3;
  bytes signature = 4;
}

message IntroductionReaction {
  oneof reaction {
    Challenge challenge = 1;
    Rejection rejection = 2;
  }
}

message Challenge {
  bytes otp = 1;
  bytes signature = 2;
}

message Rejection {}

message ChallengeAnswer {
  bytes signature = 1;
}

message AccessPass {
	bytes signature = 1;
	bool passes = 2;
}
